msal_extensions/cache_lock.py

```diff
@@ -0,0 +1,40 @@
+import os
+import sys
+import errno
+import datetime
+import portalocker
+
+
+class CrossPlatLock(object):
+    """
+    Offers a mechanism for waiting until another process is finished interacting with a shared resource. This is
+    specifically written to interact with a class of the same name in the .NET extensions library.
+    """
+    TIMEOUT = datetime.timedelta(minutes=1)
+    RETRY_WAIT = datetime.timedelta(milliseconds=100)
+    RETRY_COUNT = int(TIMEOUT.total_seconds() / RETRY_WAIT.total_seconds())
+
+    def __init__(self, lockfile_path):
+        self._lockpath = lockfile_path
+
+    def __enter__(self):
+        pid = os.getpid()
+        lock_dir = os.path.dirname(self._lockpath)
+        if not os.path.exists(lock_dir):
+            os.makedirs(lock_dir)
```

Comment:
I can understand your motivation here. But then it is argurably also needed for the real token cache file. So another approach is we move this part of logic to their common caller, the actual TokenCache class in your subsequent PRs.

Reply:
Fixed in 1731a6a